Makefile

```diff
@@ -6,15 +6,29 @@ all: init format lint
 check: format lint
 
 init:
-	ls
+	@command -v pdm >/dev/null 2>&1 || (echo "PDM not installed. Installing..." && pip install pdm && echo "PDM is installed!")
```

Comment:
We need to check python version first, before installing other libraries. The below error occurs if the version `<3.11`.

<img width="1009" alt="image" src="https://github.com/WOOSHIK-M/Quant/assets/44994859/695b864d-8fd7-4114-843a-23b194a83fc3">

we can do this with the below.

```
PYTHON := python
MIN_PYTHON_VERSION := 3.11
CURRENT_PYTHON_VERSION := $(shell $(PYTHON) -c 'import sys; print(".".join(map(str, sys.version_info[:2])))')

check-python-compatiability: 
	@echo "Checking Python version..."
	@if [ "$(CURRENT_PYTHON_VERSION)" \< "$(MIN_PYTHON_VERSION)" ]; then \
        echo "Error: Python version $(CURRENT_PYTHON_VERSION) is lower than required version $(MIN_PYTHON_VERSION)."; \
        exit 1; \
    else \
        echo "Python version $(CURRENT_PYTHON_VERSION) is compatible."; \
    fi
```